sdk/core/azure-core/azure/core/tracing/decorator.py

```diff
@@ -50,18 +58,20 @@ def wrapper_use_tracer(self, *args, **kwargs):
         ans = None
         if parent_span is not None and orig_wrapped_span is None:
             common.set_span_contexts(parent_span)
-            name = name_of_span or self.__class__.__name__ + "." + func.__name__
+            name = (
+                name_of_span or getattr(func, "__qualname__", None) or common.get_function_and_class_name(func, *args)
```

Comment:
why not put the getattr check in `get_function_and_class_name` and just call that only?

Reply:
DONE